pkg/scheduling/hostportusage_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"fmt"
+	"net"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("HostPortUsage", func() {
+	Context("entry type UT", func() {
+		It("String() output", func() {
+			ipVal := net.IPv4(10, 0, 0, 0)
+			portVal := int32(4443)
+			protocolVal := v1.ProtocolTCP
+			e := entry{
+				ip:       ipVal,
+				port:     portVal,
+				protocol: protocolVal,
+			}
+			Expect(e.String()).To(Equal(fmt.Sprintf("IP=%s Port=%d Proto=%s", ipVal, portVal, protocolVal)))
+		})
+		It("identical entries match", func() {
+			ipVal := net.IPv4(10, 0, 0, 0)
+			portVal := int32(4443)
+			protocolVal := v1.ProtocolTCP
+			e1 := entry{
+				ip:       ipVal,
+				port:     portVal,
+				protocol: protocolVal,
+			}
+			e2 := e1
+			Expect(e1.matches(e2)).To(BeTrue())
+			Expect(e2.matches(e1)).To(BeTrue())
+		})
+		It("if any one IP has an unspecified IPv4 or IPv6 address, they match", func() {
```

Comment:
I understand that the general pattern in karpenter-core isn't to target these functions for UT and test behaviors and outcomes at higher abstraction layers, but this `type entry` `match()` method was so non-intuitive to me upon doing a boring code audit that I thought it would benefit from explicit test cases to solidify intent. 

The code comment in the func is helpful, though!

> // If IPs are unequal, they don't match unless one is an unspecified address "0.0.0.0" or the IPv6 address "::".